lib/ansible/modules/network/aci/aci_l3out.py

```diff
@@ -0,0 +1,313 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l3out
+short_description: Manage Layer 3 Outside (L3Out) objects (l3ext:Out)
+description:
+- Manage Layer 3 Outside (L3Out) on Cisco ACI fabrics.
+notes:
+- The C(tenant) and C(domain) and C(vrf) used must exist before using this module in your playbook.
+  The M(aci_tenant) and M(aci_domain) and M(aci_vrf) modules can be used for this.
+- More information about the internal APIC class B(l3ext:Out) from
+  L(the APIC Management Information Model reference,https://developer.cisco.com/docs/apic-mim-ref/).
+author:
+- Rostyslav Davydenko (@rost-d)
+version_added: '2.6'
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    required: yes
+    aliases: [ tenant_name ]
+  l3out:
+    description:
+    - Name of L3Out being created.
+    required: yes
+    aliases: [ l3out_name, name ]
+  vrf:
+    description:
+    - Name of the VRF being associated with the L3Out.
+    required: yes
+    aliases: [ vrf_name ]
+  domain:
+    description:
+    - Name of the external L3 domain being associated with the L3Out.
+    required: yes
+    aliases: [ ext_routed_domain_name, routed_domain ]
+  dscp:
+    description:
+    - The target Differentiated Service (DSCP) value.
+    choices: [ AF11, AF12, AF13, AF21, AF22, AF23, AF31, AF32, AF33, AF41, AF42, AF43, CS0, CS1, CS2, CS3, CS4, CS5, CS6, CS7, EF, VA, unspecified ]
+    default: unspecified
+    aliases: [ target ]
+  route_control:
+    description:
+    - Route Control enforcement direction. The only allowed values are export or import,export.
+    choices: [ 'export', 'import' ]
+    aliases: [ route_control_enforcement ]
+  l3protocol:
+    description:
+    - Routing protocol for the L3Out
+    choices: [ static, bgp, ospf, pim ]
+  description:
+    description:
+    - Description for the L3Out.
+    default:
+    aliases: [ descr ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    choices: [ absent, present ]
+    default: present
+extends_documentation_fragment: aci
+'''
+
+EXAMPLES = r'''
```

Comment:
We prefer also examples where we remove objects, query a single object or query all objects. Look at some of the other modules. We have at minimum 4 examples.

Reply:
Thanks !